### ISSUE TITLE
Host headers are not being returned by API for workload healthcheck

### ISSUE THREAD

**Issue by tfiduccia** [rancher/rancher#16853]
Version - 2.2 master 12/4

**Steps to reproduce (least amount of steps as possible):**
1.  Create Workload
2.  Go to Advanced Options and select Health Check
3.  Select HTTP or HTTPS request radio option
4.  Fill out info and put in 1 or more host headers for readiness and liveness check
5.  Launch Workload

**Result:**  Host headers are being passed in on post but not returned by API.  So host headers are not being saved.  Same thing happens with editing and trying to add them.

**Comment by loganhz** [comment]
It happens in v2.1.1 as well. reported by @shinebayar-g

**Comment by shinebayar-g** [comment]
Oops, I've opened #16998 Duplicate issue there.

**Comment by davidnuzik** [comment]
Version: master 2.2 (12/10/18)
Confirmed fixed. I followed the steps to reproduce and could not. I verified by inspecting HTTP POST (request payload and server response) that we indeed sent our customized readiness AND aliveness checks and that the server confirmed. I made sure to set a custom host URI and additional arguments (such as User-Agent with arbitrary value). Further, confirmed that we can do this with an upgrade (editing workload) and this worked as well. Both aliveness and readiness were tested in both new workload and upgrade scenario.
